daffodil-test-integration/src/test/scala/org/apache/daffodil/cliTest/TestCLITdml.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.cliTest
+
+import org.apache.daffodil.cli.Main.ExitCode
+import org.apache.daffodil.cli.cliTest.Util._
+import org.junit.Test
+
+class TestCLITdml {
+
+  @Test def test_CLI_Tdml_Trace_singleTest1(): Unit = {
+    val tdml = path(
+      "daffodil-test/src/test/resources/org/apache/daffodil/section06/entities/Entities.tdml"
+    )
+
+    val envs = Map("DAFFODIL_TDML_API_INFOSETS" -> "all")
+
+    runCLI(args"test -i -t $tdml byte_entities_6_08", envs = envs) { cli =>
+      // legacy parse
+      cli.expect("parser: <Element name='e3'>")
+      // sax parse
+      cli.expect("parser: <Element name='e3'>")
+      cli.expect("[Pass] byte_entities_6_08")
+    }(ExitCode.Success)
+  }
+
+  @Test def test_CLI_Tdml_Trace_singleTest2(): Unit = {
+    val tdml = path(
+      "daffodil-test/src/test/resources/org/apache/daffodil/section06/entities/Entities.tdml"
+    )
+
+    val envs = Map("DAFFODIL_TDML_API_INFOSETS" -> "scala")
+
+    runCLI(args"test -i -t $tdml byte_entities_6_08", envs = envs) { cli =>
```

Comment:
Just a note that this doesn't really check that we are  running only the scala API. Unfortunately which APIs are used are way behind the scenes so there isn't really a good way to verify it, and there's no good way for the expect library to expect that a string *doesn't* exist. This test is probably still worthwhile since I imagine it gives us coverage and at least confirms that "scala" is legal, but if we accidentally break things so "scala" runs both APIs, this test probably won't fail.

Reply:
Yea I figured, there might be a way by checking the numbered output of the trace but that would break if we changed the output of trace. Thoughts?

```
----------------------------------------------------------------- 7
parser: <Element name='e3'><DelimiterStackParser>...</DelimiterStackParser></Element>
...
----------------------------------------------------------------- 17
parser: <Element name='e3'><DelimiterStackParser>...</DelimiterStackParser></Element>

```

Reply:
I don't think there's anything that can be done. Just pointing out that we need to be careful if we ever make changes to how to run single/multiple APIs in TDML, since tests tests might not tell us if we broke it.